TODO.md

```diff
@@ -0,0 +1,8 @@
+- [x] SDL should extend type for external types - I guess marking types in SDL
+  - [ ] can't generate graphql-js stuff, don't want to do it for externs - don't support graphql-js for this?
+- [ ] all imported types (so support interfaces etc)
+- [ ] Read SDL to actually do validation
+  - [ ] reenable global validations
+- [ ] "modular" mode? like no full schema, but parts of schema but with full validation by resolving it?
```

Comment:
Copy pasting from chat:

> So I'm realizing it's not rational to expect Grats to own emitting the final GraphQLSchema object if you have external types. Those types will have resolvers that Grats can't know about. So, I think if you use @gqlExternal we must assume that Grats is going to be producing a schema slice, and the consumer will be responsible for stiching it togehter with the other schemas. That could be done with resolver maps or graphql-tools schema merge.

In short, I think as soon as you use `@gqlExternal` you are implicitly opting into modular mode. We will validate against the full schema, but only emit SDL/resolver maps/GraphQLSchema for the part of the schema that Grats owns.

Reply:
I think I have a setup that should work - you can only have gqlExternal in resolver mode and it does full validation, but only generates for local types.